doc/mgmt/SONiC_IntfRangeFeature_HLD.md

```diff
@@ -0,0 +1,235 @@
+# Interface Range Specification support
+Implement support for Interface range Specification via KLISH CLI in SONiC management framework.
+
+# High Level Design Document
+#### Rev 0.1
+
+# Table of Contents
+* [List of Tables](#list-of-tables)
+* [Revision](#revision)
+* [About This Manual](#about-this-manual)
+* [Scope](#scope)
+* [Definition/Abbreviation](#definitionabbreviation)
+
+# List of Tables
+[Table 1: Abbreviations](#table-1-abbreviations)
+# Revision
+| Rev | Date | Author | Change Description |
+|:---:|:----------:|:-----------------------:|----------------------------------------------|
+| 0.1 | 05/17/2020 | Tejaswi Goel | Initial version |
+
+# Scope
+* Implementation details, syntax, and unit test cases regarding interface range command support via KLISH CLI.
+* Interface range specification will not be supported via REST/gNMI.
+* Breakout command will not be supported in interface range configuration mode.
+
+# Definition/Abbreviation
+NA
+
+# 1 Feature Overview
+
+Interface Range Specification feature support via Klish CLI will allow specification of range of interfaces to which the subsequent commands will be applied. Configuring a range of interfaces will help reduce the time and effort in configuring interfaces.
+
+## 1.1 Requirements
+
+### 1.1.1 Functional Requirements
+
+Provide ability to create/configure/show/delete range of interfaces via SONiC KLISH CLI.
+
+### 1.1.2 Configuration and Management Requirements
+* Support for *interface range* command to configure the range of interfaces in both native and alias mode.
+* Extend show and delete interface commands to support range of interfaces.
+
+### 1.1.3 Scalability Requirements
+NA
+### 1.1.4 Warm Boot Requirements
+NA
+## 1.2 Design Overview
+### 1.2.1 Basic Approach
+
+- KLISH XML changes for adding new CLI view for interface range command.
+- Use of VAR tag in KLISH XML to cache existing interfaces list.
+- Changes in KLISH actioner script to invoke APIs for range of interfaces.
+- Use **interface range** command to configure existing interfaces in the range.
+- Use **interface range create** command to create non-existing interfaces in the range.
+
+### 1.2.2 Container
+This feature is implemented within the Management Framework container.
+
+### 1.2.3 SAI Overview
+# 2 Functionality
+## 2.1 Target Deployment Use Cases
+## 2.2 Functional Description
+# 3 Design
+## 3.1 Overview
+* Creation and configuration of interfaces to be done as a single transaction.
+* Transaction for show and delete of interfaces will be per interface.
```

Comment:
Any reason for delete operation being per interface. Will it not become same as deleting per interface and thus loosing the purpose of the range command?

Reply:
delete requires the URI to have the key, so have to do 1 at a time.